### PR TITLE
Cloud regions

### PR DESCRIPTION
**Pull request by arturrez**
resolves https://github.com/ava-labs/avalanche-cli/issues/1262 
resolves https://github.com/ava-labs/avalanche-cli/issues/1138
resolves https://github.com/ava-labs/avalanche-cli/issues/1283

### CODE REVIEW

File: cmd/nodecmd/create_aws.go [hunk 1]
```diff
@@ -97,8 +98,15 @@ func getAWSCloudConfig(awsProfile string) (map[string]*awsAPI.AwsCloud, map[stri
 	ec2SvcMap := map[string]*awsAPI.AwsCloud{}
 	amiMap := map[string]string{}
 	numNodesMap := map[string]int{}
+	// verify regions are valid
+	if validRegions, err := checkRegions(maps.Keys(finalRegions)); err != nil {
+		return nil, nil, nil, err
+	} else if !validRegions {
+		return nil, nil, nil, fmt.Errorf("invalid regions provided for %s", constants.AWSCloudService)
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:105]: can you point out to the user the given invalid regions?

File: cmd/nodecmd/create_aws.go [hunk 2]
```diff
@@ -341,3 +349,21 @@ func addCertToSSH(certName string) error {
 	utils.SetupRealtimeCLIOutput(cmd, true, true)
 	return cmd.Run()
 }
+
+// checkRegions checks if the given regions are available in AWS.
+func checkRegions(regions []string) (bool, error) {
+	regionChecker, err := getAWSCloudCredentials(awsProfile, "us-east-1")
```
Comment (felipemadero) [inline cmd/nodecmd/create_aws.go:355]: the return of getAWSCloudCredentials is not a regionChecker object

Comment (felipemadero) [reply cmd/nodecmd/create_aws.go:355]: us-east-1 can be made a constant

File: pkg/cloud/gcp/gcp.go [hunk 3]
```diff
@@ -407,6 +409,49 @@ func (c *GcpCloud) StopGCPNode(nodeConfig models.NodeConfig, clusterName string)
...
+// ListZonesInRegion returns a list of zones in a specific region for a given project ID.
+func (c *GcpCloud) ListZonesInRegion(region string) ([]string, error) {
+	zoneListCall := c.gcpClient.Zones.List(c.projectID)
+	zoneList, err := zoneListCall.Do()
+	if err != nil {
+		return nil, err
+	}
+	zones := []string{}
+	for _, zone := range zoneList.Items {
+		if zone.Region == fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s", c.projectID, region) {
```
Comment (felipemadero) [inline pkg/cloud/gcp/gcp.go:435]: can the fmt string be made a constant

File: cmd/nodecmd/create_aws.go (hunk 1 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:105]: fixed

File: pkg/cloud/gcp/gcp.go (hunk 3 above)
Comment (arturrez) [reply pkg/cloud/gcp/gcp.go:435]: done

File: cmd/nodecmd/create_aws.go (hunk 2 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go:355]: it's a object that we only use to make AWS API call to check region. I am open to a better name. 
moved us-east-1 to the const

File: cmd/nodecmd/create_aws.go (hunk 1 above)
Comment (felipemadero) [reply cmd/nodecmd/create_aws.go]: I mean, to only print the invalid regions, not all the provided. I mean for checkRegions to not return a boolean, but the list of invalid regions

File: cmd/nodecmd/create_aws.go (hunk 2 above)
Comment (felipemadero) [reply cmd/nodecmd/create_aws.go]: call it awsCloud probably, it is more directly related to the data type

File: pkg/cloud/gcp/gcp.go (hunk 3 above)
Comment (felipemadero) [reply pkg/cloud/gcp/gcp.go]: in general when I talk about adding a constant. I suggest to add it to the constants package, or to constants header section of the file, or to a constants file inside the same directoty, but not to a constant defined in the same function
maybe we can agree on which style to use for this

File: cmd/nodecmd/create_aws.go (hunk 1 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go]: fixed

File: cmd/nodecmd/create_aws.go (hunk 2 above)
Comment (arturrez) [reply cmd/nodecmd/create_aws.go]: fixed name

File: pkg/cloud/gcp/gcp.go (hunk 3 above)
Comment (arturrez) [reply pkg/cloud/gcp/gcp.go]: k sorry. will fix it

**Review by felipemadero — APPROVED** [review]

**Pull request merged**